crates/cdk/src/wallet/mod.rs

```diff
@@ -123,26 +123,26 @@ impl Wallet {
     /// use cdk_sqlite::wallet::memory;
     /// use cdk::nuts::CurrencyUnit;
     /// use cdk::wallet::Wallet;
-    /// use rand::Rng;
+    /// use rand::{random, Rng};
     ///
     /// async fn test() -> anyhow::Result<()> {
-    ///     let seed = rand::thread_rng().gen::<[u8; 32]>();
+    ///     let seed = Arc::new(random::<[u8; 32]>());
     ///     let mint_url = "https://testnut.cashu.space";
     ///     let unit = CurrencyUnit::Sat;
     ///
     ///     let localstore = memory::empty().await?;
-    ///     let wallet = Wallet::new(mint_url, unit, Arc::new(localstore), &seed, None);
+    ///     let wallet = Wallet::new(mint_url, unit, Arc::new(localstore), seed, None);
     ///     Ok(())
     /// }
     /// ```
     pub fn new(
         mint_url: &str,
         unit: CurrencyUnit,
         localstore: Arc<dyn WalletDatabase<Err = database::Error> + Send + Sync>,
-        seed: &[u8],
+        seed: Arc<[u8]>,
```

Comment:
I think wallet can still just take a `&[u8]` and doesn't need the `Arc`

Reply:
Unfortunately `&[u8]` needs a lifetime specifier in

https://github.com/cashubtc/cdk/blob/ad55818288b307aaf95b3f0bb5bcc7ee380cb2a6/crates/cdk/src/wallet/multi_mint_wallet.rs#L28-L34

which then also makes `MultiMintWallet::new` need one

https://github.com/cashubtc/cdk/blob/ad55818288b307aaf95b3f0bb5bcc7ee380cb2a6/crates/cdk/src/wallet/multi_mint_wallet.rs#L36-L42

and forces the caller to deal with it in every `MultiMintWallet` call, for example in the CLI:

https://github.com/cashubtc/cdk/blob/ad55818288b307aaf95b3f0bb5bcc7ee380cb2a6/crates/cdk-cli/src/sub_commands/melt.rs#L25-L28

I would much rather keep it as `Arc<[u8]>`, as it's simpler and avoids all this.

Reply:
What do you think? Would you agree?

Reply:
Good idea, it worked: 73ae20952aadc876fbe15efca126126da7ad8791